### PR TITLE
Csv import/update master

### PR DESCRIPTION
**Pull request by michellekunz**
### CATEGORY

Choose one

- [ ] Bug Fix
- [X] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [X] Build / Development Environment
- [ ] Documentation

### SUMMARY
merge appache master in our master

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/assets/src/CsvToDatabase/CsvToDatabase.jsx [hunk 1]
```diff
@@ -340,6 +352,22 @@ export class CsvToDatabase extends React.PureComponent {
                         />
                       </td>
                     </tr>
+                    <tr>
+                      <td className="col-lg-2">
+                        {t('Table Exists')} <Asterisk />
+                      </td>
+                      <td>
+                        <FormSelect
+                          id={'tableExists'}
```
Comment (sloypog) [inline superset/assets/src/CsvToDatabase/CsvToDatabase.jsx]: Shouldn't this be "ifTableExists"?

File: superset/db_engine_specs/base.py [hunk 2]
```diff
@@ -387,7 +387,7 @@ def df_to_sql(cls, df: pd.DataFrame, **kwargs):  # pylint: disable=invalid-name
 
     @classmethod
     def create_and_fill_table_from_csv(
-        cls, form_data: MultiDict, csv_filename: str, database: Database
+        cls, form_data: MultiDict, csv_filename: str, database
```
Comment (sloypog) [inline superset/db_engine_specs/base.py:390]: shouldn't this work when using models.Database?

File: tests/csv_upload_tests.py [hunk 3]
```diff
@@ -362,19 +369,18 @@ def test_check_and_save_csv(self):
...
-    def test_create_table_in_superset(self):
+    def test_create_table_in_superset_fail(self):
         table_name = "newTable"
         example_db = utils.get_example_database()
-        table = self.importer._create_table_in_superset(table_name, example_db, None)
-        assert table.table_name == table_name
-        assert table.database == example_db
+        error_message = f"Table {table_name} could not be created."
+        with self.assertRaisesRegex(TableCreationException, error_message):
+            self.importer._create_table_in_superset(table_name, example_db, None)
```
Comment (sloypog) [inline tests/csv_upload_tests.py:377]: Why should this fail now?

File: tests/dict_import_export_tests.py [hunk 4]
```diff
@@ -282,13 +282,14 @@ def test_export_datasource_ui_cli(self):
...
-            ui_export["databases"][0]["tables"], cli_export["databases"][0]["tables"]
-        )
+
+        cli_export_database_names = {database.get("database_name") for database in cli_export["databases"]}
+        cli_export_tables = []
+        for database in cli_export["databases"]:
+            for table in database["tables"]:
+                cli_export_tables.append(table["table_name"])
+        self.assertIn(ui_export["databases"][0]["database_name"], cli_export_database_names)
+        self.assertIn(ui_export["databases"][0]["tables"][0]["table_name"], cli_export_tables)
```
Comment (sloypog) [inline tests/dict_import_export_tests.py]: Why do we need to change superset tests?

**Review by sloypog — APPROVED** [review]

File: superset/db_engine_specs/base.py (hunk 2 above)
Comment (michellekunz) [reply superset/db_engine_specs/base.py:390]: Database and model.Database are failed

**Pull request merged**